### PR TITLE
Add new enum type for ServiceOwnershipCheck ContactType

### PR DESCRIPTION
**Pull request by maxim-opslevel**

### CODE REVIEW

File: enum.go
Comment (maxim-opslevel) [inline enum.go:119]: `ANY` was the main reason this needed to be its own type. Its a valid enum value for the service ownership check but not for a regular contact type.

File: enum.go
Comment (eochoa260) [inline enum.go:141]: Should this not be here then if it's not a valid enum value on this contact type?

File: check.go
Comment (eochoa260) [inline check.go:54]: I'll defer to Kyle since I'm still not 100% clear on what needs to be a pointer or not but I think we've only set them on the Create and Update types for other implementations.

File: enum.go
Comment (maxim-opslevel) [reply enum.go:141]: ah thank you, I accidentally left that in

Comment (maxim-opslevel) [reply enum.go]: removed in latest commit

File: check.go
Comment (rocktavious) [reply check.go:54]: It needs to be a pointer if the graphql field is "allowed to be nullable"

**Review by rocktavious — CHANGES_REQUESTED** [review]
Is there a need to change any test fixtures?

Please to add a `changie` entry for the refactor.

File: check.go
Comment (maxim-opslevel) [reply check.go:289]: It currently works with this, I didn't actually change this pointer. Should I change it? Every single place we have an optional `*bool` we include `omitempty`.

**Comment by maxim-opslevel** [comment]
> Is there a need to change any test fixtures?

No

> Please to add a changie entry for the refactor.

Just did.

File: check.go
Comment (rocktavious) [reply check.go:289]: Ya this is something we recently learned.
If left like this it means that setting this field to nil will never get to the API and thus you can never "unset" this value.  This become a problem with Terraform.

Comment (maxim-opslevel) [reply check.go:289]: Should we make these changes to every instance of this at once, in a separate MR?

Comment (rocktavious) [reply check.go:289]: that fine by me

Comment (maxim-opslevel) [reply check.go:289]: In that case is there anything else needed before approval?

Comment (rocktavious) [reply check.go:289]: Sorry - will approve now

**Review by rocktavious — APPROVED** [review]
LGTM

**Pull request merged**
